library/oneview_switch.py

```diff
@@ -16,51 +16,38 @@
 # limitations under the License.
 ###
 
-from ansible.module_utils.basic import *
-try:
-    from hpOneView.oneview_client import OneViewClient
-    from hpOneView.exceptions import HPOneViewException
 
-    HAS_HPE_ONEVIEW = True
-except ImportError:
-    HAS_HPE_ONEVIEW = False
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['stableinterface'],
+                    'supported_by': 'curated'}
 
 DOCUMENTATION = '''
 ---
 module: oneview_switch
 short_description: Provides an interface to remove ToR Switch resources.
 description:
-    - "Provides an interface to remove Top of Rack(ToR) Switch resources.
-     The switch resource will be removed if it is in an unmanaged state.
-     If the switch resource is associated with a Logical Switch, it's removal is treated as a hardware removal only.
-     A reference to the switch is mantained, and the resource is marked as 'Absent'."
+    - Provides an interface to remove Top of Rack(ToR) Switch resources.
+      The switch resource will be removed if it is in an unmanaged state.
+      If the switch resource is associated with a Logical Switch, it's removal is treated as a hardware removal only.
+      A reference to the switch is mantained, and the resource is marked as 'Absent'.
+version_added: "2.3"
 requirements:
     - "python >= 2.7.9"
     - "hpOneView >= 3.0.0"
 author: "Bruno Souza (@bsouza)"
 options:
-    config:
-      description:
-        - Path to a .json configuration file containing the OneView client configuration.
-          The configuration file is optional. If the file path is not provided, the configuration will be loaded from
-          environment variables.
-      required: false
     state:
         description:
-            - Indicates the desired state for the Switch.
-              'absent' will remove the resource from OneView, if it exists.
-              'ports_updated' will update the switch ports.
+            - Indicates the desired state for the Switch resource.
+              C(absent) will remove the resource from OneView, if it exists.
+              C(ports_updated) will update the switch ports.
         choices: ['absent', 'ports_updated']
     name:
       description:
         - Switch name.
-      required: True
-notes:
-    - "A sample configuration file for the config parameter can be found at:
-       https://github.com/HewlettPackard/oneview-ansible/blob/master/examples/oneview_config-rename.json"
-    - "Check how to use environment variables for configuration at:
-       https://github.com/HewlettPackard/oneview-ansible#environment-variables"
-    - This resource is only available on C7000 enclosures
```

Comment:
This note should be kept: "-    - This resource is only available on C7000 enclosures"